app/assets/javascripts/discourse/components/topic-admin-menu-button.js.es6

```diff
@@ -7,5 +7,23 @@ export default MountWidget.extend({
 
   buildArgs() {
     return this.getProperties("topic", "fixed", "openUpwards", "rightSide");
+  },
+
+  toggleAdminMenu() {
+    $(".toggle-admin-menu")
+      .first()
+      .click();
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    this.appEvents.on("topic:toggleAdminMenu", this, this.toggleAdminMenu);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+
+    this.appEvents.off("topic:toggleAdminMenu", this, this.toggleAdminMenu);
```

Comment:
Can we instead use `this.dispatch` to send the event to the widget, which can then toggle itself rather than using a jQuery click event?

Reply:
Sure, I'll make the change right away.

Reply:
I'm having some trouble understanding how to implement `this.dispatch`. There is a method called `showAdminMenu` in the `topic-admin-menu-button` widget. This method shows the menu by changing the state `expanded`. But I don't know how to trigger that method when executing the keyboard shortcut.

Reply:
So here's an example from the site header. In didInsertElement you add `this.dispatch`:

https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/components/site-header.js.es6#L206

In this case, when `header:keyboard-trigger` is triggered as an app event, it will be dispatched to the `header` widget:

https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/widgets/header.js.es6#L533

the method name `headerKeyboardTrigger` is made automatically from `header:keyboard-trigger`

So in your case you want to use the same pattern. In your equivalent method to `headerKeyboardTrigger` you want to toggle expanded. Does that make sense?

Reply:
Yeah! It makes a lot of sense. I didn't know about the method name made automatically, that helps a lot! Thank you!